File: src_original/opengl/libqopenglversionprofile.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qsurfaceformat_enums = @import("../libqsurfaceformat.zig").enums;
pub const struct_i32_i32 = extern struct { first: i32, second: i32 };

/// https://doc.qt.io/qt-6/qopenglversionprofile.html
pub const qopenglversionprofile = struct {
    /// New constructs a new QOpenGLVersionProfile object.
    ///
    ///
    pub fn New() QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new();
    }

    /// New2 constructs a new QOpenGLVersionProfile object.
    ///
    /// ``` format: QtC.QSurfaceFormat ```
    pub fn New2(format: ?*anyopaque) QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new2(@ptrCast(format));
    }

    /// New3 constructs a new QOpenGLVersionProfile object.
    ///
    /// ``` other: QtC.QOpenGLVersionProfile ```
    pub fn New3(other: ?*anyopaque) QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new3(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#operator-eq)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile, rhs: QtC.QOpenGLVersionProfile ```
    pub fn OperatorAssign(self: ?*anyopaque, rhs: ?*anyopaque) void {
        qtc.QOpenGLVersionProfile_OperatorAssign(@ptrCast(self), @ptrCast(rhs));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#version)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    pub fn Version(self: ?*anyopaque) struct_i32_i32 {
        const _pair: qtc.libqt_pair = qtc.QOpenGLVersionProfile_Version(@ptrCast(self));
        return struct_i32_i32{
            .first = @as(*i32, @ptrCast(@alignCast(_pair.first))).*,
            .second = @as(*i32, @ptrCast(@alignCast(_pair.second))).*,
        };
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#setVersion)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile, majorVersion: i32, minorVersion: i32 ```
    pub fn SetVersion(self: ?*anyopaque, majorVersion: i32, minorVersion: i32) void {
        qtc.QOpenGLVersionProfile_SetVersion(@ptrCast(self), @intCast(majorVersion), @intCast(minorVersion));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#profile)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    ///
    /// Returns: ``` qsurfaceformat_enums.OpenGLContextProfile ```
    pub fn Profile(self: ?*anyopaque) i32 {
        return qtc.QOpenGLVersionProfile_Profile(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#setProfile)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile, profile: qsurfaceformat_enums.OpenGLContextProfile ```
    pub fn SetProfile(self: ?*anyopaque, profile: i32) void {
        qtc.QOpenGLVersionProfile_SetProfile(@ptrCast(self), @intCast(profile));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#hasProfiles)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    pub fn HasProfiles(self: ?*anyopaque) bool {
        return qtc.QOpenGLVersionProfile_HasProfiles(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#isLegacyVersion)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    pub fn IsLegacyVersion(self: ?*anyopaque) bool {
        return qtc.QOpenGLVersionProfile_IsLegacyVersion(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#isValid)
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.QOpenGLVersionProfile_IsValid(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qopenglversionprofile.html#dtor.QOpenGLVersionProfile)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QOpenGLVersionProfile ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLVersionProfile_Delete(@ptrCast(self));
    }
};

File: src_original/opengl/libqopenglversionprofile.cpp
#include <QOpenGLVersionProfile>
#include <QPair>
#include <QSurfaceFormat>
#include <qopenglversionprofile.h>
#include "libqopenglversionprofile.h"
#include "libqopenglversionprofile.hxx"

QOpenGLVersionProfile* QOpenGLVersionProfile_new() {
    return new QOpenGLVersionProfile();
}

QOpenGLVersionProfile* QOpenGLVersionProfile_new2(const QSurfaceFormat* format) {
    return new QOpenGLVersionProfile(*format);
}

QOpenGLVersionProfile* QOpenGLVersionProfile_new3(const QOpenGLVersionProfile* other) {
    return new QOpenGLVersionProfile(*other);
}

void QOpenGLVersionProfile_OperatorAssign(QOpenGLVersionProfile* self, const QOpenGLVersionProfile* rhs) {
    self->operator=(*rhs);
}

libqt_pair /* tuple of int and int */ QOpenGLVersionProfile_Version(const QOpenGLVersionProfile* self) {
    QPair<int, int> _ret = self->version();
    // Convert QPair<> from C++ memory to manually-managed C memory
    int* _first = static_cast<int*>(malloc(sizeof(int)));
    int* _second = static_cast<int*>(malloc(sizeof(int)));
    *_first = _ret.first;
    *_second = _ret.second;
    libqt_pair _out;
    _out.first = static_cast<void*>(_first);
    _out.second = static_cast<void*>(_second);
    return _out;
}

void QOpenGLVersionProfile_SetVersion(QOpenGLVersionProfile* self, int majorVersion, int minorVersion) {
    self->setVersion(static_cast<int>(majorVersion), static_cast<int>(minorVersion));
}

int QOpenGLVersionProfile_Profile(const QOpenGLVersionProfile* self) {
    return static_cast<int>(self->profile());
}

void QOpenGLVersionProfile_SetProfile(QOpenGLVersionProfile* self, int profile) {
    self->setProfile(static_cast<QSurfaceFormat::OpenGLContextProfile>(profile));
}

bool QOpenGLVersionProfile_HasProfiles(const QOpenGLVersionProfile* self) {
    return self->hasProfiles();
}

bool QOpenGLVersionProfile_IsLegacyVersion(const QOpenGLVersionProfile* self) {
    return self->isLegacyVersion();
}

bool QOpenGLVersionProfile_IsValid(const QOpenGLVersionProfile* self) {
    return self->isValid();
}

void QOpenGLVersionProfile_Delete(QOpenGLVersionProfile* self) {
    delete self;
}
